.github/workflows/publish.yml

```diff
@@ -0,0 +1,40 @@
+name: Build and publish
+
+on:
+  push:
+    branches-ignore:
+      - 'main'
+      - '*backport*'
+    tags:
+      - 'v*'
+      - '!*dev*'
+      - '!*pre*'
+      - '!*post*'
+  pull_request:
+    branches:
+      - 'main'
```

Comment:
I'm guessing this will go once we've checked the build is working in this PR?

Reply:
Yes, that will be removed before merging. Do we want to include the lines below which should always build on PRs if they modify this workflow file?

Reply:
That sounds like a good idea

Reply:
oh that's a cool little feature.